src/JMSSerializerAdapter.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Liip\Serializer\Adapter\JMS;
+
+use JMS\Serializer\ArrayTransformerInterface;
+use JMS\Serializer\Context as JMSContext;
+use JMS\Serializer\DeserializationContext as JMSDeserializationContext;
+use JMS\Serializer\SerializationContext as JMSSerializationContext;
+use JMS\Serializer\SerializerInterface;
+use Liip\Serializer\Context;
+use Liip\Serializer\SerializerInterface as LiipSerializer;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Bridge to use the LiipSerializer as a drop-in replacement for JMS
+ * Serializer, with fallback to JMS on error.
+ *
+ * If the liip serializer can not handle the operation, we fall back to the
+ * regular JMS serializer.
+ */
+class JMSSerializerAdapter implements SerializerInterface, ArrayTransformerInterface
+{
+    /**
+     * A hashmap with FQN classnames that should be handled with the generated
+     * serializer or null to always try to find a generated serializer.
+     *
+     * @var bool[]|null
+     */
+    private $enabledClasses;
+
+    /**
+     * The original serializer *MUST* implement both SerializerInterface and ArrayTransformerInterface interfaces
+     *
+     * @var SerializerInterface&ArrayTransformerInterface Fallback for when the type or format is not supported
+     */
+    private $originalSerializer;
+
+    /**
+     * @var LiipSerializer
+     */
+    private $liipSerializer;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param SerializerInterface&ArrayTransformerInterface $originalSerializer must implement both SerializerInterface and ArrayTransformerInterface interfaces
+     * @param string[]|null                                 $enabledClasses     list of fully qualified class names for which to use the generated serializer.
+     *                                                                          Null tells to attempt to use this serializer with all classes.
+     */
+    public function __construct(
+        LiipSerializer $liipSerializer,
+        $originalSerializer,
+        LoggerInterface $logger,
+        ?array $enabledClasses = null
+    ) {
+        $this->liipSerializer = $liipSerializer;
+        if (!$originalSerializer instanceof SerializerInterface
+            || !$originalSerializer instanceof ArrayTransformerInterface
```

Comment:
@ChristianRiesen this is where phpstan complains (the thing i put into ignore). it seems reasonable to me to do this, there is no parameter type declaration because we require the jms serializer to implement both its interfaces.

Reply:
Seems to me that this could be a false positive in phpstan, maybe something that we can open an issue for over there.